debezium-core/src/main/java/io/debezium/relational/RelationalSnapshotChangeEventSource.java

```diff
@@ -246,10 +247,12 @@ protected abstract void readTableStructure(ChangeEventSourceContext sourceContex
     /**
      * Releases all locks established in order to create a consistent data snapshot.
      */
-    protected void releaseDataSnapshotLocks(RelationalSnapshotContext<O> snapshotContext) throws Exception {
+    protected void releaseDataSnapshotLocks(P partition, RelationalSnapshotContext<O> snapshotContext) throws Exception {
```

Comment:
Same question as above: how about having a snapshot change event source per partition instead? Alternatively, if we want to keep a single SCES instance, exposing the partition via the `snapshotContext` seems good, it already contains mutable state as the snapshot is progressing.

Reply:
Sounds good. I moved the partition argument into the state of the object that contains the corresponding offset context at the last iteration but apparently not everywhere. Will do.